internal/index/indexoperations/index.go

```diff
@@ -52,3 +53,14 @@ func ListIndexes(path string) ([]Index, error) {
 	}
 	return indexes, nil
 }
+
+// AddIndex initializes a new index to install plugins from.
+func AddIndex(path, name, url string) error {
+	dir := filepath.Join(path, name)
```

Comment:
@corneliusweig wdyt of a preliminary check to see if url is an actual url? Simple url.Parse can help.

Reply:
I thought this is not desirable, because it might break magic corp URLs with non-standard protocol prefixes. Also, somebody pointed out that `url.Parse` is not foolproof, although I don't remember exactly why.

IMHO, `EnsureCloned` should surface a readable error message? Maybe we also want to delete the index directory if cloning failed. Otherwise users will have to manually delete the index directory.

Reply:
If bad URLs are handled by `git`, maybe we need to be more careful that the user actually sees an error. Can you make sure of that?

Reply:
The git clone error does ultimately get wrapped and surfaced to the user. Is that enough or did you mean to add some other context around the git error?

Reply:
Ok, that is what we need. This should be good enough.